### PR TITLE
Persist contract store

### PR DESCRIPTION
**Pull request by MixailE**

### CODE REVIEW

File: src/helpers/ContractSynchronizer.ts [hunk 1]
```diff
@@ -5,15 +5,64 @@ import getOwnedERC721, {
...
+    account: string,
+    addressToTokenIds: { [address: string]: Set<string> } = {},
+    synchronizedBlockId?: number
+  ) {
     this.account = account
+    this.synchronizedBlockId = synchronizedBlockId
+    this.addressToTokenIds = addressToTokenIds
+  }
+
+  static fromJSON({
```
Comment (T-Damer) [inline src/helpers/ContractSynchronizer.ts:31]: Let's name it `contractSynchronizerFromJson`

File: src/helpers/ContractSynchronizer.ts [hunk 2]
```diff
@@ -5,15 +5,64 @@ import getOwnedERC721, {
...
+      {}
+    )
+    return new ContractSynchronizer(
+      account,
+      addressToTokenSetIds,
+      synchronizedBlockId
+    )
+  }
+
+  toJSON() {
```
Comment (T-Damer) [inline src/helpers/ContractSynchronizer.ts:54]: And this is `contractSynchronizerToJson`

File: src/stores/ContractsStore.ts [hunk 3]
```diff
@@ -1,64 +1,73 @@
...
-    if (!ContractsStore.connectedAccounts[WalletStore.account])
-      ContractsStore.connectedAccounts[WalletStore.account] =
-        new ContractSynchronizer(WalletStore.account)
-    if (
-      accountChange ||
-      !ContractsStore.connectedAccounts[WalletStore.account] ||
-      !ContractsStore.contractsOwned
-    ) {
-      ContractsStore.contractsOwned = ContractsStore.connectedAccounts[
+    if (!this.connectedAccounts[WalletStore.account]) {
```
Comment (T-Damer) [inline src/stores/ContractsStore.ts:45]: You can do assignment without `{}`

File: src/stores/ContractsStore.ts [hunk 4]
```diff
@@ -1,64 +1,73 @@
...
+}
+
+const contractsStore = proxyWithComputed(new ContractsStore(), {
+  contractsOwned: (state) =>
+    walletStore.account && state.connectedAccounts[walletStore.account]
+      ? state.connectedAccounts[walletStore.account].contractsOwned
+      : [],
+}).makePersistent(true)
 
 subscribeKey(WalletStore, 'account', () =>
```
Comment (T-Damer) [inline src/stores/ContractsStore.ts:64]: Maybe we can pass `subscribeKey(WalletStore, 'account', contractStore.fetchMoreContractsOwned)`?

File: src/stores/ContractsStore.ts [hunk 5]
```diff
@@ -1,64 +1,73 @@
...
-  contractsOwned: Promise.resolve([]),
-  currentBlock: undefined,
+  replacer = (key: string, value: unknown) => {
+    const disallowList = [] as string[] // ['contractsOwned']
+    return disallowList.includes(key) ? undefined : value
+  }
+
+  reviver = (key: string, value: unknown) => {
+    if (key === 'connectedAccounts') {
+      return Object.entries(
```
Comment (T-Damer) [inline src/stores/ContractsStore.ts:21]: Why do we need to convert the object into `array-like` object?

File: src/helpers/ContractSynchronizer.ts [hunk 6]
```diff
@@ -5,15 +5,64 @@ import getOwnedERC721, {
...
+    return {
+      account: this.account,
+      synchronizedBlockId: this.synchronizedBlockId,
+      addressToTokenIds: Object.entries(this.addressToTokenIds).reduce(
```
Comment (T-Damer) [inline src/helpers/ContractSynchronizer.ts:58]: It's pretty hard to understand on what is going on here, maybe you can simplify it?

File: src/helpers/ContractSynchronizer.ts [hunk 7]
```diff
@@ -5,15 +5,64 @@ import getOwnedERC721, {
...
+  static fromJSON({
+    account,
+    synchronizedBlockId,
+    addressToTokenIds,
+  }: {
+    account: string
+    synchronizedBlockId: number
+    addressToTokenIds: { [address: string]: string[] }
+  }) {
+    const addressToTokenSetIds = Object.entries(addressToTokenIds).reduce(
```
Comment (T-Damer) [inline src/helpers/ContractSynchronizer.ts:40]: It's pretty hard to understand on what is going on here, maybe you can simplify it? Maybe we can extract it into a function?

**Review by T-Damer — CHANGES_REQUESTED** [review]
looks good, let's simplify it :)

File: src/helpers/ContractSynchronizer.ts (hunk 2 above)
Comment (MixailE) [reply src/helpers/ContractSynchronizer.ts:54]: I see no reason to add this before the method name, this is a similar method as `toString`, `toNumber`, `toObject`

File: src/stores/ContractsStore.ts (hunk 4 above)
Comment (MixailE) [reply src/stores/ContractsStore.ts:60]: Here we can’t do this, the context is lost

File: src/helpers/ContractSynchronizer.ts (hunk 1 above)
Comment (MixailE) [reply src/helpers/ContractSynchronizer.ts:32]: I see no reason to add this before the method name, this is a similar method as `toString`, `toNumber`, `toObject`

File: src/helpers/ContractSynchronizer.ts (hunk 6 above)
Comment (MixailE) [reply src/helpers/ContractSynchronizer.ts]: Added `transformObjectValues` ​​function :)

File: src/helpers/ContractSynchronizer.ts (hunk 7 above)
Comment (MixailE) [reply src/helpers/ContractSynchronizer.ts]: Added `transformObjectValues` ​​function :)

**Review by T-Damer — APPROVED** [review]

**Review by backmeupplz — APPROVED** [review]

**Pull request merged**